sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/BuilderHelper.java

```diff
@@ -85,6 +91,20 @@ static HttpPipeline buildPipeline(Supplier<HttpPipelinePolicy> credentialPolicyS
             .build();
     }
 
+    /**
+     * Sets the allowed headers and queries to logOptions.
+     * @param logOptions the log options for headers and queries.
+     */
+    private static void loadLogOptions(final HttpLogOptions logOptions) {
+        Objects.requireNonNull(logOptions);
+
+        logOptions.setAllowedHeaderNames(StorageAllowedHeadersAndQueries.FileShareHeadersAndQueries
+            .getFileShareHeaders());
+
+        logOptions.setAllowedQueryParamNames(StorageAllowedHeadersAndQueries.FileShareHeadersAndQueries
```

Comment:
Are you intending to overwrite the existing white list with your white list, or are you intended to supplement the existing white list with additional headers / query params? At present you're overwriting, which may not be what you're intending to do.

Reply:
Good point. I need to append to existing ones.